src/bt_auto_dumper/_v2/__main__.py

```diff
@@ -53,15 +49,34 @@ def dump_and_upload(subnet_identifier: str, autovalidator_address: str, api_key:
     with zipfile.ZipFile(zip_filename, "w") as zipf:
         for file in output_files:
             zipf.write(file)
-    send_to_autovalidator(zip_filename, autovalidator_address, api_key)
+    send_to_autovalidator(zip_filename, wallet, autovalidator_address, note, subnet_identifier)
+
 
+def make_signed_request(method, url, headers, files, wallet):
+    headers["Nonce"] = str(time.time())
+    headers["Hotkey"] = wallet.hotkey.ss58_address
+    file_content = files.get("file").read()
+    files.get("file").seek(0)
+    headers_str = json.dumps(headers, sort_keys=True)
+    data_to_sign = f"{method}{url}{headers_str}{file_content}".encode()
+    signature = wallet.hotkey.sign(
+        data_to_sign,
+    ).hex()
+    headers["Signature"] = signature
 
-def send_to_autovalidator(zip_filename, autovalidator_address, api_key):
+    response = requests.request(method, url, headers=headers, files=files)
+    return response
+
+
+def send_to_autovalidator(zip_filename, wallet, autovalidator_address, note, subnet_identifier):
     url = f"{autovalidator_address}/api/v1/files/"
-    headers = {"Authorization": f"Token {api_key}"}
     files = {"file": open(zip_filename, "rb")}
 
-    response = requests.post(url, headers=headers, files=files)
+    headers = {
+        "Note": note,
+        "SubnetID": ",".join(subnet_identifier),
```

Comment:
```suggestion
        "SubnetID": subnet_identifier,
```
I think this is going to produce `c,o,m,p,u,t,e,h,o,r,d,e`